File: src/minecraft/net/minecraft/client/gui/GuiMainMenu.java
package net.minecraft.client.gui;


import me.fontloader.FontDrawer;
import me.skrilled.SenseHeader;
import me.skrilled.ui.alt.GuiAltLogin;
import me.skrilled.utils.render.RenderUtil;
import me.surge.animation.Animation;
import me.surge.animation.ColourAnimation;
import me.surge.animation.Easing;
import net.minecraft.client.main.Main;
import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.resources.I18n;

import java.awt.*;
import java.io.IOException;

public class GuiMainMenu extends GuiScreen implements GuiYesNoCallback {

    Animation animation = new Animation(1500f, false, Easing.CUBIC_OUT);
    Animation newA=new Animation(1000f,false,Easing.CUBIC_OUT);
    ColourAnimation colourAnimation = new ColourAnimation(new Color(0, 0, 0, 30), new Color(0, 0, 0, 100), 1500f, false, Easing.LINEAR);

    public void initGui() {
        int interval = 30;
        int buttonWidth = 100;
        int buttonHeight = 20;
        int j = this.height / 2 - buttonHeight / 2;
        this.buttonList.add(new GuiButton(1, interval, j + interval * 2, buttonWidth, buttonHeight, I18n.format("menu.quit"), animation, animation));
        this.buttonList.add(new GuiButton(2, interval, j + interval, buttonWidth, buttonHeight, I18n.format("menu.options"), animation, animation));
        this.buttonList.add(new GuiButton(3, interval, j, buttonWidth, buttonHeight, "Alt:" + mc.session.getUsername(), animation, animation));
        this.buttonList.add(new GuiButton(4, interval, j - interval, buttonWidth, buttonHeight, I18n.format("menu.multiplayer"), animation, animation));
        this.buttonList.add(new GuiButton(5, interval, j - interval * 2, buttonWidth, buttonHeight, I18n.format("menu.singleplayer"), animation, animation));
    }


    protected void actionPerformed(GuiButton button) throws IOException {

        if (button.id == 5) mc.displayGuiScreen(new GuiSelectWorld(this));

        if (button.id == 4) mc.displayGuiScreen(new GuiMultiplayer(this));

        if (button.id == 3) mc.displayGuiScreen(new GuiAltLogin(this));

        if (button.id == 2) mc.displayGuiScreen(new GuiOptions(this, mc.gameSettings));

        if (button.id == 1) SenseHeader.getSense.stopClient();

    }

    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        animation.setState(isHovering(mouseX, mouseY, 0, 0, width / 4f, height));
        colourAnimation.setState(isHovering(mouseX, mouseY, 0, 0, width / 4f, height));        newA.setState(isHovering(mouseX, mouseY, 0, 0, width / 4f, height));

        this.drawBackground();
        int j = this.height / 2;
        FontDrawer font = Main.fontLoader.EN48;

        GlStateManager.pushMatrix();


        int interval = 30;
        int buttonWidth = 100;
        RenderUtil.drawRect(0, 0, (buttonWidth+interval*2) * animation.getAnimationFactor(), height*animation.getAnimationFactor(), colourAnimation.getColour().getRGB());
        font.drawRainbowString(SenseHeader.getSense.getClientName(), 30, j - 90);
        GlStateManager.popMatrix();

        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    private boolean isHovering(int mouseX, int mouseY, float xLeft, float yUp, float xRight, float yBottom) {
        return mouseX > xLeft && mouseX < xRight && mouseY > yUp && mouseY < yBottom;
    }

    protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
        super.mouseClicked(mouseX, mouseY, mouseButton);
    }


}

File: minecraft/me/kl0udy92/apart/screen/menu/GuiMainMenu.kt
package me.kl0udy92.apart.screen.menu

import me.kl0udy92.apart.core.Main
import me.kl0udy92.apart.screen.menu.account.menu.GuiAccountManager
import me.kl0udy92.apart.screen.menu.button.GuiSimpleButton
import me.kl0udy92.apart.utils.render.MouseUtil
import me.kl0udy92.apart.utils.render.RenderUtil
import me.kl0udy92.apart.utils.render.shaders.implementations.MainMenuShader
import net.minecraft.client.gui.*
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.util.ResourceLocation
import org.lwjgl.input.Mouse
import org.lwjgl.opengl.GL11
import org.lwjgl.opengl.GL20
import java.awt.Color

class GuiMainMenu(
    private val sandboxShader: MainMenuShader = MainMenuShader(0),
    private val languagesLocation: ResourceLocation = ResourceLocation("${Main.name.lowercase()}/icons/menu/languages.png"),
    private val settingsLocation: ResourceLocation = ResourceLocation("${Main.name.lowercase()}/icons/menu/settings.png")
) : GuiScreen() {

    override fun initGui() {
        this.simpleButtonList.add(GuiSimpleButton(0, this.width / 2 - 100, this.height / 2 - 42, "Singleplayer"))
        this.simpleButtonList.add(GuiSimpleButton(1, this.width / 2 - 100, this.height / 2 - 21, "Multiplayer"))
        this.simpleButtonList.add(GuiSimpleButton(2, this.width / 2 - 100, this.height / 2, "Shut down"))
        super.initGui()
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        /*mc.textureManager.bindTexture(ResourceLocation("apart/background.jpg"))
        drawModalRectWithCustomSizedTexture(
            -mouseX,
            -mouseY,
            0f,
            0f,
            width * 2, height * 2,
            width * 2F, height * 2.toFloat()
        )*/
        this.drawBackground()
        this.renderIconButtons(mouseX, mouseY)
        super.drawScreen(mouseX, mouseY, partialTicks)
    }

    override fun actionPerformed(button: GuiButton) {
        when (button.id) {
            0 -> mc.displayGuiScreen(GuiSelectWorld(this))
            1 -> mc.displayGuiScreen(GuiMultiplayer(this))
            2 -> mc.shutdown()
        }
        super.actionPerformed(button)
    }

    fun drawBackground() {
        GlStateManager.enableAlpha()
        GlStateManager.disableCull()
        this.sandboxShader.render(width.toFloat(), height.toFloat())
        GL11.glBegin(GL11.GL_QUADS)

        GL11.glVertex2f(-1f, -1f)
        GL11.glVertex2f(-1f, 1f)
        GL11.glVertex2f(1f, 1f)
        GL11.glVertex2f(1f, -1f)

        GL11.glEnd()

        GL20.glUseProgram(0)
    }

    private fun renderIconButtons(mouseX: Int, mouseY: Int) {
        val isHoveringAltsLocation = MouseUtil.isHovered(5, 5, 24, 24, mouseX, mouseY)
        val isHoveringLanguagesLocation = MouseUtil.isHovered(width - 24, height - 24, 24, 24, mouseX, mouseY)
        val isHoveringSettingsLocation = MouseUtil.isHovered(width - 24, 0, 24, 24, mouseX, mouseY)
        if (isHoveringAltsLocation) {
            RenderUtil.drawRect(4, 4, 5 + 24 + 1, 5 + 24 + 1, Color.BLACK.rgb)
            if (Mouse.isButtonDown(0)) mc.displayGuiScreen(GuiAccountManager(this))
        }
        RenderUtil.drawFace(this.mc.session.profile, 5, 5)
        RenderUtil.drawImage(
            this.languagesLocation,
            width - 24f,
            height - 24f,
            24f,
            24f,
            if (isHoveringLanguagesLocation) Color(243, 233, 224).rgb else Color(200, 200, 200).rgb
        )
        if (isHoveringLanguagesLocation && Mouse.isButtonDown(0))
            mc.displayGuiScreen(GuiLanguage(this, mc.gameSettings, mc.languageManager))
        RenderUtil.drawImage(
            settingsLocation,
            (width - 24).toFloat(),
            0f,
            24f,
            24f,
            if (isHoveringSettingsLocation) Color(243, 233, 224).rgb else Color(200, 200, 200).rgb
        )
        if (isHoveringSettingsLocation && Mouse.isButtonDown(0))
            mc.displayGuiScreen(GuiOptions(this, mc.gameSettings))
    }

}